File: shared/src/commonJvm/kotlin/logger/SentryNapierIntegration.kt
import io.github.aakira.napier.Napier
import io.sentry.*

class SentryNapierIntegration(
    val minEventLevel: SentryLevel = SentryLevel.ERROR,
    val minBreadcrumbLevel: SentryLevel = SentryLevel.INFO
) : Integration {
    override fun register(hub: IHub, options: SentryOptions) {
        val antiLog = SentryAntiLog(
            hub = hub,
            minEventLevel = minEventLevel,
            minBreadcrumbLevel = minBreadcrumbLevel
        )
        Napier.base(antiLog)
    }

}

File: iosApp/iosApp/Logger/SentryNapierIntegration.swift
//
//  SentryNapierIntegration.swift
//  iosApp
//
//  Created by Carlos Monzon on 10/8/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import Sentry
import ComposeApp

class SentryNapierIntegration : NSObject, SentryIntegrationProtocol {
    
    let minEventLevel: SentryLevel = SentryLevel.error
    let minBreadcrumbLevel: SentryLevel = SentryLevel.info
    
    func install(with options: Options) -> Bool {
        let antiLog = SentryAntiLog(minEventLevel: minEventLevel, minBreadcumbLevel: minBreadcrumbLevel)
        LogKt.releaseBuild(antilog: antiLog)
        return true
    }
    
    func uninstall() {}
}
